utils/umaApi.ts

```diff
@@ -0,0 +1,103 @@
+import type { ContentfulSynth } from "./contentful";
+
+const baseOptions = {
+  headers: {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+  },
+  method: "POST",
+};
+
+const baseUrl = process.env.UMA_API_URL || "http://dev.api.umaproject.org:8282";
+
+function constructRequest(...params: unknown[]) {
+  try {
+    return { ...baseOptions, body: JSON.stringify([...params]) };
+  } catch (e) {
+    throw new Error(`Could not stringify ${params}`);
+  }
+}
+
+export async function request<T>(
+  method: string,
+  ...params: unknown[]
+): Promise<T> {
+  const response = await fetch(
+    `${baseUrl}/${method}`,
+    constructRequest(...params)
+  );
+  if (!response.ok) {
+    throw new Error(
+      JSON.stringify({ status: response.status, message: response.statusText })
+    );
+  }
+  return response.json();
+}
+
+export type EmpState = {
+  id: string;
+  address: string;
+  priceIdentifier: string;
+  expirationTimestamp: string;
+  withdrawLiveness: string;
+  tokenCurrency: string;
+  collateralCurrency: string;
+  finder: string;
+  minSponsorTokens: string;
+  liquidationLiveness: string;
+  collateralRequirement: string;
+  disputeBondPercentage: string | null;
+  sponsorDisputeRewardPercentage: string | null;
+  disputerDisputeRewardPercentage: string | null;
+  cumulativeFeeMultiplier: string;
+  updated: number;
+  totalTokensOutstanding: string;
+  totalPositionCollateral: string;
+  rawTotalPositionCollateral: string;
+  expiryPrice: string;
+  sponsors: string[];
+  tokenDecimals: number;
+  collateralDecimals: number;
+  tokenName: string;
+  collateralName: string;
+  tokenSymbol?: string;
+  collateralSymbol?: string;
+  gcr: string;
+  identifierPrice: string;
+};
```

Comment:
this looks really annoying, wish there was a better way to get this

Reply:
Agree! I'm sure there are some packages that "introspec" an API and extract types but typing manually was faster so I just went with it